### PR TITLE
feat: Add GDPO Support

### PR DESCRIPTION
**Pull request by ved1beta**
used trl  `multi_objective_aggregation: (
        Literal["sum_then_normalize", "normalize_then_sum"]` https://github.com/huggingface/trl/pull/4785

* **Tests**
  * Added comprehensive end-to-end test suite for GDPO across multi-GPU setups with diverse configurations.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/axolotl/core/trainers/grpo/trainer.py [hunk 1]
```diff
@@ -64,6 +64,7 @@ class AxolotlGRPOTrainer(
     """Extend the base GRPOTrainer for axolotl helpers"""
 
     _tag_names = ["trl", "grpo", "axolotl"]
+    _last_loaded_step: int
```
Comment (NanoCode012) [inline src/axolotl/core/trainers/grpo/trainer.py]: What's this for?

File: docs/rlhf.qmd [hunk 2]
```diff
@@ -720,6 +721,102 @@ trl:
...
+        - rewards.correctness_reward
+    reward_weights: [1.0, 2.0]
+
+datasets:
+    - path: openai/gsm8k
+      name: main
+      type: rewards.oai_gsm8k_transform
+```
+
+You can also use GRPO with explicit aggregation control:
```
Comment (NanoCode012) [inline docs/rlhf.qmd:763]:
```suggestion
You can also use GDPO with explicit aggregation control:
```
typo?

Edit: Wait, I noticed this part uses `grpo`, so unsure if intended.

**Review by winglian — CHANGES_REQUESTED** [review]
requirements.txt needs to be updated to `trl==0.27.0` in order for the test to actually work I think.

File: src/axolotl/utils/schemas/trl.py [hunk 3]
```diff
@@ -179,3 +179,13 @@ class TRLConfig(BaseModel):
             "description": "Path to custom rollout function. Must be importable from current dir."
         },
     )
+    multi_objective_aggregation: (
```
Comment (winglian) [inline src/axolotl/utils/schemas/trl.py:182]: in `RLValidationMixin` we should add a check that if `rl: gdpo` is set and `multi_objective_aggregation` is set incorrectly, it should error

File: tests/e2e/multigpu/solo/test_gdpo.py [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+from accelerate.test_utils import execute_subprocess_async
+from transformers.testing_utils import get_torch_dist_unique_port
+
+from axolotl.utils.dict import DictDefault
+
+from tests.e2e.multigpu.solo.test_grpo import recursive_kill, start_vllm
+from tests.e2e.utils import require_vllm
+
+
+@pytest.mark.skip(reason="flaky vllm tests in modal")
```
Comment (winglian) [inline tests/e2e/multigpu/solo/test_gdpo.py:23]: did we test that these are actually flaky in modal? or is this just copy paste from the GRPO test suite?

**Review by winglian — APPROVED** [review]
good to go. We can re-enable the vllm+g[rd]po tests after the vllm upgrade pr lands
